src/python/pants/engine/goal.py

```diff
@@ -70,7 +76,17 @@ def subsystem_dependencies(cls):
       return dep,
     return tuple()
 
-  options_scope_category = ScopeInfo.GOAL
+  @classmethod
+  def is_implemented(cls, *, union_membership: "UnionMembership") -> bool:
```

Comment:
Is it possible to simplify this a bit and have a default-empty `required_unions = [..]` property? I think that that would allow for better error messages out of the box (because you could say something like "goal $ requires a backend with support for $union_docstring"), and might be harder to implement incorrectly?

Reply:
Fantastic recommendation! Thank you!